napari/utils/key_bindings.py

```diff
@@ -182,6 +187,257 @@ def normalize_key_combo(key_combo):
+        command: callable
+            take 0, or 1 parameter; is `instance` is not None, will be called
+            `instance` as first parameter.
+        description: str
+            Long string to describe what the command does, will be used in
+            tooltips.
+
+        """
+        assert name not in self._actions, name
+        self._actions[name] = Action(command, description, keymappable)
+        self._update_shortcut_bindings(name)
+        # shortcuts may have been bound before the command was actually
+        # registered, remove it from orphan and bind shortcuts.
+        if command in self._orphans:
+            # print('Found orphan shortcut')
+            sht = self._orphans.pop(command)
+            self.bind_shortcut(name, sht)
+        self._update_gui_elements(name)
+
+    def _update_gui_elements(self, name):
+        """
+        Update the description and shortcuts of all the (known) gui elements.
+        """
+        if name not in self._actions:
+            return
+        buttons = self._buttons[name]
+        desc = self._actions[name].description
+
+        # update buttons with shortcut and description
+        if name in self._shortcuts:
+            sht = self._shortcuts[name]
+            sht_str = f' ({format_shortcut(sht)})'
+        else:
+            sht = ''
+            sht_str = ''
+        for button in buttons:
+            button.setToolTip(desc + sht_str)
+        if name in self._qactions:
+            action = self._actions[name]
+            qaction = self._qactions[name]
+            qaction.setShortcut(
+                QKeySequence(sht.replace('-', '+').replace('Control', 'Ctrl'))
+            )
+            qaction.setText(action.description)
+            qaction.setStatusTip(action.description)
+
+    def _update_shortcut_bindings(self, name):
+        """
+        Update the key mappable for given action name
+        to trigger the action within the given context and
+        make all the corresponding UI element flash if possible.
+        """
+        if name not in self._actions:
+            return
+        action = self._actions[name]
+        if name not in self._shortcuts:
+            return
+        sht = self._shortcuts.get(name)
+        keymappable = action.keymappable
+        if hasattr(keymappable, 'bind_key'):
+
+            def flash(*args):
+                call_with_context(action.command, self.context)
+                for b in self._buttons.get(name, []):
+                    b._animation.start()
+
+            keymappable.bind_key(sht)(flash)
+
+    def bind_button(self, name, button):
+        """
+        Bind `button` to trigger Action `name` on click.
+        """
+        action = self._actions[name]
+
+        button.clicked.connect(
+            lambda: call_with_context(action.command, self.context)
+        )
+
+        self._buttons[name].append(button)
+        self._update_gui_elements(name)
+
+    def bind_shortcut(self, name, shortcut):
```

Comment:
I think we could make the shortcut be part of the action creation/definition, by default. It feels a bit redundant to have this method if we could have this bound directly on the action creation?